ipa-core/src/report/hybrid.rs

```diff
@@ -80,6 +214,119 @@ where
     }
 }
 
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct EncryptedHybridImpressionReport<BK, B>
+where
+    B: Deref<Target = [u8]>,
+    BK: SharedValue,
+{
+    data: B,
+    phantom_data: PhantomData<BK>,
+}
+
+impl<BK, B> EncryptedHybridImpressionReport<BK, B>
+where
+    B: Deref<Target = [u8]>,
+    BK: SharedValue,
+    Replicated<BK>: Serializable,
+    <Replicated<BK> as Serializable>::Size: Add<U16>,
+    <<Replicated<BK> as Serializable>::Size as Add<U16>>::Output: ArrayLength,
+{
+    const ENCAP_KEY_MK_OFFSET: usize = 0;
```

Comment:
same here. If the only reason for these bounds is to be able to write `const E: X + Y + ... + Replicated<BA64>::Size::USIZE` instead of `const E = X + Y + Replicated<BA>::size()`, then it is not worth it imo

Reply:
Note that you may want https://github.com/private-attribution/ipa/pull/1357

Reply:
That would be cool. I'll look into that when it lands :)

Reply:
> same here. If the only reason for these bounds is to be able to write const E: X + Y + ... + Replicated<BA64>::Size::USIZE instead of const E = X + Y + Replicated<BA>::size(), then it is not worth it imo

agreed. in fact, I don't think there is any advantage at all to having these be known at compile time. even with all these bounds, we can still submit bad input which will cause an error.

Reply:
That PR is merged, so we can follow up on it

Reply:
After playing around with this a little, it looks like I can't use ::size() when defining a type. This means that I need the bounds to implement Serializable (to define Size type), as well as to implement decryption (to define CTMKLength and CTBTTLength<BK> types). Then for encryption, when I try and remove the Add bounds, it has a problem with SITE_DOMAIN_OFFSET, even when I use size() to define the constants leading up to that.